### PR TITLE
bugfix: TCC support cross-service

### PR DESCRIPTION
**Pull request by zhongfuhua**
### Ⅰ. Describe what this PR did
Dubbo transaction ID cannot be passed in TCC transaction

### Ⅳ. Describe how to verify it
Use the Dubbo service and test the scenario as transaction - > Service 1 - > service 2

### CODE REVIEW

File: spring/src/main/java/io/seata/spring/tcc/TccActionInterceptor.java [hunk 1]
```diff
@@ -39,7 +40,7 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(TccActionInterceptor.class);
 
-    private static final String DUBBO_PROXY_NAME_PREFIX="com.alibaba.dubbo.common.bytecode.proxy";
+    private static final String DUBBO_PROXY_NAME_PREFIX = "com.alibaba.dubbo.common.bytecode.proxy";
```
Comment (jsbxyyx) [inline spring/src/main/java/io/seata/spring/tcc/TccActionInterceptor.java]: Do you need to consider apache dubbo version?

Comment (zhongfuhua) [reply spring/src/main/java/io/seata/spring/tcc/TccActionInterceptor.java]: see TransactionPropagationFilter.java

File: core/src/main/java/io/seata/core/context/RootContext.java [hunk 2]
```diff
@@ -36,6 +37,8 @@
      */
     public static final String KEY_XID = "TX_XID";
 
+    public static final String KEY_XID_INTTERCEPTOR_TYPE = "TX_XID_INTTERCEPTOR_TYPE";
```
Comment (xingfudeshi) [inline core/src/main/java/io/seata/core/context/RootContext.java]: interceptor

File: core/src/main/java/io/seata/core/context/RootContext.java [hunk 3]
```diff
@@ -46,7 +49,26 @@
...
     public static String getXID() {
-        return CONTEXT_HOLDER.get(KEY_XID);
+        String xid = CONTEXT_HOLDER.get(KEY_XID);
+        if (StringUtils.isNotBlank(xid)) {
+            return xid;
+        }
+
+        String xidType = CONTEXT_HOLDER.get(KEY_XID_INTTERCEPTOR_TYPE);
+        if (StringUtils.isNotBlank(xidType)) {
+            return xidType.split("_")[0];
```
Comment (zjinlei) [inline core/src/main/java/io/seata/core/context/RootContext.java]: &&indexOf("_")>-1

File: spring/src/main/java/io/seata/spring/tcc/TccActionInterceptor.java [hunk 4]
```diff
@@ -109,13 +112,13 @@ protected Method getActionInterfaceMethod(MethodInvocation invocation) {
             }
             if (interfaceType == null && remotingDesc.getInterfaceClassName() != null) {
                 interfaceType = Class.forName(remotingDesc.getInterfaceClassName(), true,
-                    Thread.currentThread().getContextClassLoader());
+                        Thread.currentThread().getContextClassLoader());
```
Comment (zjinlei) [inline spring/src/main/java/io/seata/spring/tcc/TccActionInterceptor.java]: please restore

File: spring/src/main/java/io/seata/spring/tcc/TccActionInterceptor.java [hunk 5]
```diff
@@ -109,13 +112,13 @@ protected Method getActionInterfaceMethod(MethodInvocation invocation) {
...
             }
             if (interfaceType == null) {
                 return invocation.getMethod();
             }
             Method method = interfaceType.getMethod(invocation.getMethod().getName(),
-                invocation.getMethod().getParameterTypes());
+                    invocation.getMethod().getParameterTypes());
```
Comment (zjinlei) [inline spring/src/main/java/io/seata/spring/tcc/TccActionInterceptor.java]: please restore

File: integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java [hunk 6]
```diff
@@ -62,14 +68,19 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
         } finally {
             if (bind) {
                 String unbindXid = RootContext.unbind();
+                String unbindInterceptorType = RootContext.unbindInterceptorType();
                 if (LOGGER.isDebugEnabled()) {
                     LOGGER.debug("unbind[" + unbindXid + "] from RootContext");
+                    LOGGER.debug("unbind interceptorType[" + unbindInterceptorType + "] from RootContext");
```
Comment (zjinlei) [inline integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java]: please merge log

File: integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java [hunk 7]
```diff
@@ -62,14 +68,19 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
...
                 }
                 if (!rpcXid.equalsIgnoreCase(unbindXid)) {
                     LOGGER.warn("xid in change during RPC from " + rpcXid + " to " + unbindXid);
+                    LOGGER.warn("xidInterceptorType in change during RPC from " + rpcXidInterceptorType + " to " + unbindInterceptorType);
                     if (unbindXid != null) {
                         RootContext.bind(unbindXid);
+                        RootContext.bindInterceptorType(unbindInterceptorType);
                         LOGGER.warn("bind [" + unbindXid + "] back to RootContext");
+                        LOGGER.warn("bind interceptorType [" + unbindInterceptorType + "] back to RootContext");
```
Comment (zjinlei) [inline integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java]: please merge log and user {}

File: integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java [hunk 8]
```diff
@@ -41,19 +40,24 @@
     @Override
     public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcException {
         String xid = RootContext.getXID();
+        String xidInterceptorType = RootContext.getXIDInterceptorType();
+
         String rpcXid = RpcContext.getContext().getAttachment(RootContext.KEY_XID);
+        String rpcXidInterceptorType = RpcContext.getContext().getAttachment(RootContext.KEY_XID_INTERCEPTOR_TYPE);
```
Comment (jsbxyyx) [inline integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java]: the `RootContext.KEY_XID_INTERCEPTOR_TYPE` may be appear lowcase. issue #1728 . The suggestion is to use lowercase at the beginning. `tx-xid-interceptor-type`

File: core/src/main/java/io/seata/core/context/RootContext.java [hunk 9]
```diff
@@ -36,6 +37,8 @@
      */
     public static final String KEY_XID = "TX_XID";
 
+    public static final String KEY_XID_INTERCEPTOR_TYPE = "TX-XID-INTERCEPTOR-TYPE";
```
Comment (jsbxyyx) [inline core/src/main/java/io/seata/core/context/RootContext.java]: lowercase

File: core/src/main/java/io/seata/core/context/RootContext.java [hunk 10]
```diff
@@ -46,7 +49,26 @@
      * @return the xid
      */
     public static String getXID() {
-        return CONTEXT_HOLDER.get(KEY_XID);
+        String xid = CONTEXT_HOLDER.get(KEY_XID);
+        if (StringUtils.isNotBlank(xid)) {
+            return xid;
+        }
+
+        String xidType = CONTEXT_HOLDER.get(KEY_XID_INTERCEPTOR_TYPE);
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/context/RootContext.java]: If using datasourceProxy how to make it not repeat the registration branch

Comment (slievrly) [reply core/src/main/java/io/seata/core/context/RootContext.java]: Is my fault, in the connection to determine whether it is used in the transaction inGlobalTransaction.

**Pull request merged**